File: elixir/lib/build_info.ex
defmodule Vektor.Runtime.VXL.BuildInfo do
  @derive Jason.Encoder
  defstruct git_sha: nil,
            build_timestamp: nil,
            build_semver: nil,
            profile: nil

  @type t(git_sha, build_timestamp, build_semver, profile) :: %__MODULE__{
          git_sha: git_sha,
          build_timestamp: build_timestamp,
          build_semver: build_semver,
          profile: profile
        }

  @type t :: %__MODULE__{
          git_sha: String.t(),
          build_timestamp: String.t(),
          build_semver: String.t(),
          profile: String.t()
        }
end

File: elixir/src/build_info.rs
use rustler::NifStruct;

#[derive(Debug, NifStruct)]
#[module = "VXL.BuildInfo"]
struct BuildInfo {
  git_sha: String,

  build_timestamp: String,
  build_semver: String,

  profile: String,
}

#[rustler::nif]
fn build_info() -> BuildInfo {
  BuildInfo {
    git_sha: env!("VERGEN_GIT_SHA_SHORT").to_string(),

    build_timestamp: env!("VERGEN_BUILD_TIMESTAMP").to_string(),
    build_semver: env!("VERGEN_BUILD_SEMVER").to_string(),

    profile: env!("VERGEN_CARGO_PROFILE").to_string(),
  }
}
